driver/SDL/SDL_keyboard.c

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file keyboard.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "SDL_keyboard.h"
+#if USE_KEYBOARD != 0
+
+/*********************
+ *      DEFINES
+ *********************/
+#ifndef MONITOR_ZOOM
+#define MONITOR_ZOOM    1
+#endif
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+static bool keyup = false;
+static bool keydown = false;
+static bool repeat = false;
+static bool keyboard_was_read = true;
+static int last_key = -1;
+
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+/**
+ * Get the current key and state of the keyboard
+ * @param data store the keyboard data here
+ * @return false
+ */
+void keyboard_read(struct _lv_indev_drv_t * indev_drv, lv_indev_data_t * data)
+{
+    if(!keyboard_was_read)
+    {
+        /* Store the collected data */
+        if(keyup)
+        {
+            data->state = LV_INDEV_STATE_PR;
+            data->key = last_key;
+        }
+        else if(keydown)
+        {
+            data->state = LV_INDEV_STATE_REL;
+            data->key = last_key;
+        }
+    }
+    if(keydown && repeat){
+        data->state = LV_INDEV_STATE_PR;
+        data->key = last_key;
+    }
+    
+    data->continue_reading = false;
+    keyboard_was_read = true;
+    return;
+}
+
+/**
+ * It will be called from the main SDL thread
+ */
+void keyboard_handler(SDL_Event * event)
+{
+    keydown = false;
+    keyup = false;
+    last_key = -1;
+
+    if(keyboard_was_read)
+    {
+        switch(event->type) {
+            case SDL_KEYUP:
+                    keyup = true;
+                    repeat = event->key.repeat;
+                    keyboard_was_read = false;
+                break;
+            case SDL_KEYDOWN:
+                    keydown = true;
+                    repeat = event->key.repeat;
+                    keyboard_was_read = false;
+                break;
+        }
+
+        if(keyup || keydown)
+        {
+            switch(event->key.keysym.sym)
```

Comment:
I think it would be cleaner to have a seperate function do the translation from `event->key.keysym.sym` to LVGL key.  
See [keycode_to_ctrl_key](https://github.com/lvgl/lv_drivers/blob/75088aecdb7a90d27152b49e9917c90611471c7f/sdl/sdl_common.c#L223) as an example.